libs/beacon/src/beacon_setup_service.cpp

```diff
@@ -106,6 +108,27 @@ BeaconSetupService::BeaconSetupService(MuddleInterface &muddle, Identity identit
 
   // Set subscription for receiving shares
   shares_subscription_->SetMessageHandler(this, &BeaconSetupService::OnNewSharesPacket);
+
+  // Set time slots for each state
+  static std::once_flag flag;
```

Comment:
I used static here because I only wanted time_slots_in_dkg_ to be computed once, otherwise on each creation of a beacon_setup_service it would accumulate. I could alternatively make time_slots_in_dkg_ not static in beacon_setup_service_, but the removal of this would mean each beacon_setup_service would set the values in time_slot_map_ again which is unnecessary. Constexpr does not work for the std::once_flag as it needs to be non-const to be passed into std::call_once.

Reply:
This is true, for tests there would be as many copies of this map as there are alive beacon setup services, however it is a pretty small amount of memory. You wouldn't need a call once if it was a constexpr right because by definition it would only invoke once